Assets/Tile walking mini game/Scripts/GridManager.cs

```diff
@@ -0,0 +1,167 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class GridManager : MonoBehaviour
+{
+    [SerializeField] private Tile[][] tileGrid;
+
+    [SerializeField] private TileColorManager tileColorManager;
+
+    [SerializeField] private ObstaclesPosition[][] obstacleLayouts;
+
+    public Action WinEvent;
+
+    public Action LoseEvent;
+    public int UntouchedTileCount { get; private set; }
+
+    private void Awake()
+    {
+        PopulateTileGrid();
+
+        CreateObstaclesLayouts();
+    }
+
+    private void Start()
+    {
+        ResetGrid();
+    }
+
+    public void ResetGrid()
+    {
+        foreach (Tile[] tileColumn in tileGrid)
+        {
+            foreach(Tile tile in tileColumn)
+            {
+                tile.TileStatus = TileStatus.untouched;
+            }
+        }
+
+        ObstaclesPosition[] randomlyPickedLayout = obstacleLayouts[UnityEngine.Random.Range(0, obstacleLayouts.Length)];
+
+        UntouchedTileCount = 25 - randomlyPickedLayout.Length;
+
+        AddingObsctacles(randomlyPickedLayout);
+
+        foreach (Tile[] tileColumn in tileGrid)
+        {
+            foreach (Tile tile in tileColumn)
+            {
+                tileColorManager.ChangeTileColorBasedOnStatus(tile);
+            }
+        }
+
+        ArrangingTilesInAGrid();
+
+        UpdateTileStatus(0, 0);
+    }
+
+    private void PopulateTileGrid()
+    {
+        tileGrid = new Tile[5][];

```

Comment:
again with the magic numbers

Reply:
Done. I created gridWidth and gridHeight variables